src/main/java/io/confluent/copycat/hdfs/HdfsWriter.java

```diff
@@ -62,106 +64,119 @@ public HdfsWriter(HdfsSinkConnectorConfig connectorConfig) {
       if (!fs.exists(path)) {
         fs.mkdirs(path);
       }
-      offsets = readOffsets();
+      offsets = new HashMap<>();
       writerProviderClass = (Class<RecordWriterProvider>)
-          Class.forName(connectorConfig.getString(HdfsSinkConnectorConfig.RECORD_WRITER_PROVIDER_CLASS_CONFIG));
+          Class.forName(connectorConfig.getString(
+              HdfsSinkConnectorConfig.RECORD_WRITER_PROVIDER_CLASS_CONFIG));
+
+      this.assignment = assignment;
     } catch (IOException e) {
-      throw new CopycatRuntimeException(e);
+      throw new CopycatException(e);
     } catch (ClassNotFoundException e) {
-      throw new CopycatRuntimeException("RecordWriterProvider class not found", e);
+      throw new CopycatException("RecordWriterProvider class not found", e);
     }
   }
 
-
-  public Map<TopicPartition, Long> getPreviousOffsets() {
+  /**
+   * @return get committed offsets of previous task runs.
+   */
+  public Map<TopicPartition, Long> getCommittedOffsets() {
     return offsets;
   }
 
-  public void writeRecord(TopicPartition topicPart, SinkRecord record)
-      throws CopycatException{
+  public void writeRecord(final TopicPartition topicPart, final SinkRecord record)
+      throws CopycatException {
     try {
       RecordWriter<Long, SinkRecord> writer = getWriter(topicPart, record);
       writer.write(System.currentTimeMillis(), record);
+
       if (!offsets.containsKey(topicPart)) {
-        offsets.put(topicPart, record.getOffset() - 1);
+        offsets.put(topicPart, record.kafkaOffset() - 1);
       }
       updateRecordCounter(topicPart);
       if (shouldRotate(topicPart)) {
         rotate(topicPart);
       }
     } catch (IOException e) {
-      throw new CopycatRuntimeException(e);
+      throw new CopycatException(e);
     }
   }
 
-  public void close() throws IOException, CopycatException {
-    for (TopicPartition topicPart: writers.keySet()) {
+  public void close() throws CopycatException {
+
+    for (TopicPartition topicPart : assignment) {
```

Comment:
Should we be catching any exceptions within this loop? If we're closing, I'm not sure we want a failure on the first TopicPartition to cause the rest of the process to be aborted -- we'd still like to try to close all the other topic partitions (and the WALs). Same goes for the call to `wal.close()` in the loop right below this one.

We might want to save any exceptions that occur and only rethrow them (or throw a wrapper around them) at the very end of the method.